desc/perturbations.py

```diff
@@ -371,7 +371,7 @@ def perturb(  # noqa: C901 - FIXME: break this up into simpler pieces
                 value, np.where(np.abs(value) < 10 * np.finfo(value.dtype).eps)[0], 0
             )
             # don't set nonexistent profile (values are empty ndarrays)
-            if not (key == "c_l" or key == "i_l") or value.size:
+            if value.size:
```

Comment:
why did we have that not block for `c_l` and `i_l`?

Reply:
to prevent setting current on fixed iota eq and vice verse, but its not needed since in that only `c_l` or `i_l` would have size>0, so we only need to check the size.